ingest/indexer/service/indexer_streaming_service.go

```diff
@@ -214,11 +214,16 @@ func (s *indexerStreamingService) publishTxn(ctx context.Context, req types.Requ
 	events := res.GetEvents()
 	var includedEvents []domain.EventWrapper
 	for i, event := range events {
-		err := s.adjustTokenInAmountBySpreadFactor(ctx, &event)
+		err := s.AdjustTokenInAmountBySpreadFactor(ctx, &event)
 		if err != nil {
 			s.logger.Error("Error adjusting amount by spread factor", "error", err)
 			continue
 		}
+		err = s.AddTokenLiquidity(ctx, &event)
```

Comment:
Could we make sure that we make a deep copy of the event?

That is, we do not modify the event in the `res.GetEvents`.

I'm not sure how these are used in the caller but events are state-breaking, and I'm worried that if this modification is propagated to the SDK, we might have a consensus failure.

Reply:
VERY GOOD POINT! 
Done.